Tests/Security/Authentication/GoogleApiAuthenticationTest.php

```diff
@@ -0,0 +1,144 @@
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Class GoogleApiAuthenticationTest
+ *
+ * Tests the Authentication of the Google API.
+ *
+ * @package HP\Bundle\EmailApiBundle\Tests\Security\Authentication
+ */
+class GoogleApiAuthenticationTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var GoogleApiAuthentication
+     */
+    private $googleApiAuthentication;
+
+    /**
+     * @var GoogleResourceOwner
+     */
+    private $googleResourceOwner;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var string
+     */
+    private $firewallName;
+
+    public function setUp()
+    {
+        $this->firewallName = 'firewallNameTest';
+        $this->googleResourceOwner =
+            $this->getMockBuilder('HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\GoogleResourceOwner')
+                ->disableOriginalConstructor()
+                ->getMock();
+        $this->tokenStorage =
+            $this->getMockBuilder('Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface')
+                ->getMock();
+
+        $this->googleApiAuthentication = new GoogleApiAuthentication($this->firewallName, $this->googleResourceOwner, $this->tokenStorage);
+    }
+
+    /**
+     * Tests if the access token is refreshed.
+     */
+    public function testRefreshTokenNotExpired()
+    {
+        $today = new \DateTime();
+        $oauthToken = $this->getOAuthToken();
+        $oauthToken->setCreatedAt($today->getTimestamp());
+        $oauthToken->setExpiresIn(100);
+        $oauthTokenSerialized = serialize($oauthToken);
+        $request = $this->getRequestWithSession($oauthTokenSerialized);
+
+        $tokenHasBeenUpdated = $this->googleApiAuthentication->refreshToken($request);
+
+        $this->assertFalse($tokenHasBeenUpdated);
+    }
+
+    /**
+     * Test when the access token is expired
+     */
+    public function testRefreshTokenExpired()
+    {
+        $today = new \DateTime();
+        $oauthToken = $this->getOAuthToken();
+        $oauthToken->setCreatedAt($today->getTimestamp());
+        $oauthToken->setExpiresIn(-100);
+        $oauthTokenSerialized = serialize($oauthToken);
+        $request = $this->getRequestWithSession($oauthTokenSerialized);
+        $this->mockTokenStorage($oauthToken);
+        $this->mockGoogleResourceOwner();
+
+        $tokenHasBeenUpdated = $this->googleApiAuthentication->refreshToken($request);
+
+        $this->assertTrue($tokenHasBeenUpdated);
+    }
+
+    /**
+     * Test when the access token is expired, but there is no token available in the
+     * token storage to be updated.
+     */
+    public function testRefreshTokenExpiredWithoutTokenInStorage()
+    {
+        $today = new \DateTime();
+        $oauthToken = $this->getOAuthToken();
+        $oauthToken->setCreatedAt($today->getTimestamp());
+        $oauthToken->setExpiresIn(-100);
```

Comment:
Take this value out and create a global setting/variable. State the units on the value declaration.